libflux/flux-core/src/formatter/mod.rs

```diff
@@ -1819,6 +1824,56 @@ impl<'doc> Formatter<'doc> {
             "/",
         ]
     }
+
+    fn format_attribute_list(&mut self, n: &'doc Vec<ast::Attribute>) -> Doc<'doc> {
+        let arena = self.arena;
+        let mut doc = arena.nil();
+        for attr in n {
+            doc += self.format_attribute(attr).group();
+            doc += arena.hardline();
+        }
+        doc
+    }
+
+    fn format_attribute(&mut self, n: &'doc ast::Attribute) -> Doc<'doc> {
+        let arena = self.arena;
+        let mut doc = docs![
+            arena,
+            self.format_comments(&n.base.comments),
+            arena.text(format!("@{}", n.name)),
+        ];
```

Comment:
```suggestion
        let mut doc = docs![
            arena,
            self.format_comments(&n.base.comments),
            "@",
            &n.name,
        ];
```

Reply:
My understanding is this change will allow `@` and the name to be split from each other. The scanner grammar doesn't allow `@ attribute` so these two are a single atom. If the suggested change ensures the two will be together, then this change is fine. Otherwise, I think the previous one needs to remain.

Reply:
No, you must add softline/hardline/etc to get explicit whitespace the formatter will always lay them out as `@attribute` with the change.

Reply:
Ok updated this then.